### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by andrewbird**
Enshrine the coding style

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    else {                                                                      
+      multiple_line1();                                                         
+      multiple_line2();                                                         
+      multiple_line3();                                                         
+    }                                                                           
+  }                                                                             
+                                                                                
+  return 0;                     // no round brackets                            
+}                                                                               
+```
```
Comment (stsp) [inline CONTRIBUTING.md]: This is a coding style, no too many
newlines at the end of file. :)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Always write a clear log message for your commits. One-line messages are fine for small changes, but bigger changes should look like this:
+
+    $ git commit -m "A brief summary of the commit
+    > 
+    > A paragraph describing what changed and its impact."
+
+## Coding conventions
+Since dosemu2 has a long history, there are many different coding styles in use. We'd like to standardise on a subtle variant of 
+the Kernighan & Ritchie style, but mass conversion is not really an option as we don't want to obscure the commit history. So
```
Comment (stsp) [inline CONTRIBUTING.md]: This is a coding style, so please make
the lines shorter. :) They wrap.

Comment (andrewbird) [reply CONTRIBUTING.md]: Yes, I'll wrap them manually, I didn't consider you might look at it as a plain text file. The Markdown format is a little like html it usually has no newlines within the paragraph, then it wraps when rendering.

**Comment by andrewbird** [comment]
I hope that's better, I killed all the trailing whitespace and fixed the wrapping. I've also learnt a lesson not to use the Github Markdown editor!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+      printf("unmatched switch case 0x%02x\n", z);
+      return 1;
+  }
+
+  while (x == y) {
+    a();
+    b();
+
+    if (z)
+      single_line();
```
Comment (stsp) [inline CONTRIBUTING.md]: Lets go for linux convention that if
you have `{` at least on one part, you
should put it in a second part too.

Comment (andrewbird) [reply CONTRIBUTING.md]: okay done

**Comment by stsp** [comment]
Thanks!

**Pull request merged**